ddl/ddl_api.go

```diff
@@ -980,6 +980,91 @@ func (d *ddl) CreateTable(ctx sessionctx.Context, s *ast.CreateTableStmt) (err e
 	return errors.Trace(err)
 }
 
+func (d *ddl) CreateView(ctx sessionctx.Context, s *ast.CreateViewStmt) (err error) {
+	ident := ast.Ident{Name: s.ViewName.Name, Schema: s.ViewName.Schema}
+	is := d.GetInformationSchema(ctx)
+	schema, ok := is.SchemaByName(ident.Schema)
+	if !ok {
+		return infoschema.ErrDatabaseNotExists.GenWithStackByArgs(ident.Schema)
+	}
+	if is.TableExists(ident.Schema, ident.Name) && !s.OrReplace {
+		return infoschema.ErrTableExists.GenWithStackByArgs(ident)
+	}
+	if err = checkTooLongTable(ident.Name); err != nil {
+		return err
+	}
+	viewInfo, cols := buildViewInfoWithTableColumns(ctx, s)
+
+	for _, v := range viewInfo.Cols {
+		if len(v.L) > mysql.MaxColumnNameLength {
+			return ErrTooLongIdent.GenWithStackByArgs(v.L)
+		}
+	}
+
+	for _, v := range cols {
+		if len(v.Name.L) > mysql.MaxColumnNameLength {
+			return ErrTooLongIdent.GenWithStackByArgs(v.Name.L)
+		}
+	}
+
+	colNames := map[string]bool{}
```

Comment:
1. Can we extract a function for this check?
2. Maybe we can use set.StringSet to replace this.

Reply:
done